utils/components.py

```diff
@@ -298,3 +307,340 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+        model_registry_api_url_parsed = urllib.parse.urlparse(
+            output_model_registry_api_url
+        )
+        model_registry_api_url_port = model_registry_api_url_parsed.port
+
+        if model_registry_api_url_port:
+            model_registry_api_server_address = output_model_registry_api_url.replace(
+                model_registry_api_url_parsed.netloc,
+                model_registry_api_url_parsed.hostname,
+            )
+        else:
+            if model_registry_api_url_parsed.scheme == "http":
+                model_registry_api_url_port = 80
+            else:
+                model_registry_api_url_port = 443
+
+            model_registry_api_server_address = output_model_registry_api_url
+
+        if not model_registry_api_url_parsed.scheme:
+            model_registry_api_server_address = (
+                "https://" + model_registry_api_server_address
+            )
+
+        with open(
+            "/var/run/secrets/kubernetes.io/serviceaccount/token", "r"
+        ) as token_file:
+            token = token_file.readline()
+
+        registry = ModelRegistry(
+            server_address=model_registry_api_server_address,
+            port=model_registry_api_url_port,
+            author="InstructLab Pipeline",
+            user_token=token,
+        )
+        if registry.get_model_version(model_name, model_version) is not None:
+            print(f"""
+            ###################################################### ERROR ######################################################
+            # The version {model_version} for model {model_name} is already registered. You cannot overwrite a model version. #
+            ###################################################################################################################
+            """)
+            sys.exit(1)
+    except StoreError as store:
+        # The model has no versions registered.
+        # Do nothing, just to avoid this exception to return failure
+        print(f"""
+        ########### INFO ##############
+        # Model Registry is available #
+        ###############################
+        """)
+        sys.exit(0)
+    except Exception as e:
+        print(f"""
+        ############# ERROR ###############
+        # Model Registry is not available #
+        ###################################
+        """)
+        raise
+
+
+@dsl.component(base_image=RUNTIME_GENERIC_IMAGE)
+def test_training_operator():
+    import sys
+
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+
+    config.load_incluster_config()
+
+    with client.ApiClient() as api_client:
+        api_instance = client.CustomObjectsApi(api_client)
+
+        group = "kubeflow.org"
+        version = "v1"
+        plural = "pytorchjobs"
+
+        try:
+            api_response = api_instance.list_cluster_custom_object(
```

Comment:
As discussed in Slack, this needs to be `list_namespaced_custom_object` since PyTorchJobs are namespace scoped.